packages/fx-core/src/plugins/solution/fx-solution/v2/getQuestions.ts

```diff
@@ -394,11 +399,45 @@ export async function getQuestionsForAddCapability(
     type: "multiSelect",
     staticOptions: [],
     default: [],
+    validation: {
+      validFunc: async (input: string[]): Promise<string | undefined> => {
+        if (
+          input.includes(NotificationOptionItem.id) &&
+          input.includes(CommandAndResponseOptionItem.id)
+        ) {
+          return getLocalizedString(
+            "core.addCapabilityQuestion.notificationCommandAndResponseConflict"
+          );
+        }
+
+        // undefined for success
+        return undefined;
+      },
+    },
+    onDidChangeSelection: async function (
+      currentSelectedIds: Set<string>,
+      previousSelectedIds: Set<string>
+    ): Promise<Set<string>> {
+      return handleSelectionConflict(
+        [
+          new Set([BotOptionItem.id, MessageExtensionItem.id]),
+          new Set([NotificationOptionItem.id]),
+          new Set([CommandAndResponseOptionItem.id]),
+        ],
+        previousSelectedIds,
+        currentSelectedIds
+      );
+    },
   };
```

Comment:
nit: how about moving the validation func to a separate function instead of expanding in the `getQuestionsForAddCapability` method. Less code lines in a method's implementation will be more readable :)

Reply:
separated to functions